Framework/ICat/src/CatalogDownloadDataFiles.cpp

```diff
@@ -169,6 +169,8 @@ std::string CatalogDownloadDataFiles::doDownloadandSavetoLocalDrive(
     Poco::Net::SecureStreamSocket *socket =
         new Poco::Net::SecureStreamSocket(context);
     Poco::Net::HTTPSClientSession session(*socket);
```

Comment:
I'm still concerned that socket is not being destroyed. I tried digging deeper into the base class and it appears to be constructing a new socket. As far as I can tell the copy constructor is called in  `Poco::Net::HTTPSession` and that (different?) object is closed in that class's destructor. 

https://github.com/pocoproject/poco/blob/develop/Net/include/Poco/Net/HTTPSession.h

Why do we have to heap allocate `socket` and if so where is it deleted?